Add Crystal libraries

wout opened the pull request:
## Checklist

- [X] I have read the [How to Contribute](https://developers.cardano.org/docs/portal-contribute/).
- [X] I have read the [Builder Tool Requirements](https://github.com/cardano-foundation/developer-portal/edit/staging/src/data/builder-tools.js)
- [X] I have run `yarn build` after adding my changes **without getting any errors**. 

## Builder Tool addition

* Title: *blockfrost-crystal*
* Description: *A Crystal SDK for the Blockfrost.io API.* 
* Website:   https://github.com/blockfrost/blockfrost-crystal
* Tags:
  * `crystal`
  * `http`
  * `sdk`
  * `serialization`
___
* Title: *cardano-kit*
* Description: *At toolkit for Crystal to ease development for the Cardano blockchain.* 
* Website: https://github.com/wout/cardano-kit
* Tags:
  * `crystal`
  * `serialization`

rphair commented on src/data/builder-tools.js:
```diff
@@ -53,6 +53,15 @@ export const Tags = {
+    description:
+      "Crystal language",
+    icon: null,
+    color: '#000',
```
this colour is already in use by `chainindex` ... can you pick another one (randomly or however it suits you) that seems visually distinct from the other colours?

wout replied:
Their style guides suggest using either `#000`, `#fff` or `#777`:

https://crystal-lang.org/media/

Since white is invisible on a light background and `#777` is visually too close to **HTTP**, I've opted to use `#ddd`. It reads white-ish and it's visually distinct from anything else. It also works well on both the light as well as the dark theme:

![image](https://user-images.githubusercontent.com/107324/225241405-ff675852-3543-4cca-a53d-42e399ab5efa.png)

![image](https://user-images.githubusercontent.com/107324/225242029-e03037fe-541f-4d21-a4a2-a15c7ff316ac.png)

rphair commented on src/data/builder-tools.js:
```diff
@@ -972,14 +981,38 @@ export const Showcases = [
-  }
+  },
+  {
+    title: "bech32",
```
Since this title appears as a builder tool name and there's already a _bech32_ (`input-output-hk/bech32`), can you call it something distinct like:
- `bech32-crystal`
- `wout/bech32
?

wout replied:
Sure, I renamed it to `bech32.cr` since that's often used in the community.

rphair reviewed (commented):
thanks @wout ... I've never heard of Crystal until just now.  There are more well rounded developers than me on the review team and chances are someone is familiar with it... I can't yet justify adding a new tag (`crystal`) for something I've never heard of before, but I'm not against it either... let's see what the others on the team say at first. 🤓

I believe you will want to choose 3 new images for each project because:
- each portoflio image displays a maximum of 353px wide in desktop view so they don't get nearly this kind of clarity & people will be squinting at a blur.
- that box has a greater than 2:1 aspect ratio, so most of what makes each project distinct is cut off.
- what appears "above the fold" is really just the project name & GitHub standard tabs, which doesn't tell the viewer anything that isn't already in your project title & description.
- therefore, best strategy is to use this aspect ratio to select or fashion a **logo-like image** for each one.

rdlrt reviewed (commented):
> There are more well rounded developers than me on the review team and chances are someone is familiar with it... I can't yet justify adding a new tag (crystal) for something I've never heard of before, but I'm not against it either... let's see what the others on the team say at first.

While I full agree with the messaging, being a developer portal, I am a bit biased to allow language tags even if a single project use-case atm, thereby - allow incoming viewer to browse availability by languages. Thus, for me - that part is fine.

The only part I am unsure about is bech32[.cr] appearing as a seperate item (existing IO one as well as new one), as ways for encoding/decoding schemes seem very granular use-cases, but given that there is already one atm - I suppose I cannot discard another.

One thing I do find a bit wanting is the overall scope of the presented `cardano-kit` (again the title could possibly include crystal to keep it distinct) - currently it seems to only cater for:
1. Address Conversions
2. Message signing based on CIP-08
Are there further plans for this repository?

rdlrt commented on src/data/builder-tools.js:
```diff
@@ -972,14 +981,38 @@ export const Showcases = [
+    preview: require("./builder-tools/blockfrost-crystal.png"),
+    website: "https://github.com/blockfrost/blockfrost-crystal",
+    getstarted: null,
+    tags: ["crystal", "sdk", "json", "http"],
```
dont think SDKs qualify for `http` tag

wout replied:
Fixed.

wout commented:
@rphair 
> I can't yet justify adding a new tag (`crystal`) for something I've never heard of before

Crystal is a fairly new language but it's gaining traction, especially in the Ruby community. I'm biased of course, but when Blockfrost asked which languages needed an SDK, Crystal popped up a few times on Twitter.

> I believe you will want to choose 3 new images for each project

Good point. I've changed them and it looks a lot better now. 

@rdlrt 
> The only part I am unsure about is bech32[.cr] appearing as a seperate item

Removed. I was also in doubt about it but added it anyway since IO's Haskell lib was in there. It's used in the cardano-kit shard and I'm sure other developers will be able to find it on Crystal's shard listing sites.

> One thing I do find a bit wanting is the overall scope of the presented cardano-kit (again the title could possibly include crystal to keep it distinct)

It's basically a set of abstractions for common interactions with Cardano. Soon I'll push a new release containing a CLI wrapper for cardano-cli, which I wrote for the minting apps of DendroRithms and The Perfect NFT. It'll receive more updates in the coming months.
I get your concern about the scope. The reason I opted to make it a toolkit is that the individual components are often used together. And individually, many are too small to create separate Crystal shards.

rphair reviewed (commented):
@wout I'm ready to approve after last round of changes but noticed that `yarn.lock` snuck into this PR (normally it's `gitignore`'d) with your original commit https://github.com/cardano-foundation/developer-portal/commit/e470b74ffcc7c8cc08da616a47cee02b503afe7e.

It should be much easier for you to eradicate this than us: if needing ideas on how to restore to baseline I wrote a bit about it here: https://developers.cardano.org/docs/portal-contribute#should-i-commit-the-yarnlock-file

rphair reviewed (commented):
@wout we've gone from a PR with a modified `yarn.lock` to a PR that would remove `yarn.lock` from our `staging` branch if it were merged.  We need a PR that has a `yarn.lock` file identical to what is currently in our `staging` branch.

Even with our `.gitignore` the way it is now, an unwanted `yarn.lock` update seems an occasional consequence of committing from local builds and it's happened to me once or twice without knowing why.  This seems most likely to happen if you do a `yarn install` because it always generates a new `yarn.lock`.

Anyway, like I said before, the link above has some ideas for how to grab a current copy of `yarn.lock` from whatever `staging` branch around your fork doesn't have the latest modifications to `yarn.lock` in it.  My usual fix is to just check it out directly from `staging` in the upstream repo `cardano-foundation/developer-portal`. 🤓

wout commented:
Got you. I thought you meant it should not have been in there. I've now reverted the file to the last commit before I created the branch.
